### PR TITLE
log errors in bottle to Sentry

### PR DESCRIPTION
**Pull request by r3m0t**
Repalces #1138

### CODE REVIEW

**Comment by r3m0t** [comment]
I'm no longer at Festicket so I can't test this (well, the Sentry aspect of it)

**Comment by gbin** [comment]
We don't use sentry neither at Argo, anyone with a sentry setup can test this?

**Comment by GenPage** [comment]
@gbin We do at DO. I can test this now that 5.2.0 is released with my Sentry transport feature.

File: errbot/bootstrap.py [hunk 1]
```diff
@@ -74,6 +76,37 @@ def bot_config_defaults(config):
...
+            "You have BOT_LOG_SENTRY enabled, but I couldn't import modules "
+            "needed for Sentry integration. Did you install raven? "
+            "(See http://raven.readthedocs.org/en/latest/install/index.html "
+            "for installation instructions)"
+        )
+        exit(-1)
+
+    kwargs = {'dsn': config.SENTRY_DSN}
+
+    if hasattr(config, 'SENTRY_TRANSPORT') and isinstance(config.SENTRY_TRANSPORT, tuple):
```
Comment (gbin) [inline errbot/bootstrap.py]: Maybe a simple get is enough with a default to None?

**Comment by GenPage** [comment]
Noticed #1214 while testing this PR, might want to hold and refactor until #1214 is merged.

**Comment by gbin** [comment]
#1214 is merged (note it will be for 6.0, if you want this for 5.0 feel free to make the PR against the 5.2 branch

**Comment by gbin** [comment]
Is it ok to merge this one? Someone can rebase it and check if it is ok?

**Comment by GenPage** [comment]
@r3m0t can you rebase against the 5.2 branch instead of master please?

**Comment by GenPage** [comment]
@r3m0t Awesome, will do!

**Comment by andrewthetechie** [comment]
@GenPage any test results on this PR?

**Comment by sijis** [comment]
Does this still work work/apply. I thought we moved off of `bottle` and we are using flask instead. This  leads me to think that too https://github.com/errbotio/errbot/blob/8c3d74995e316a14870527901ce89f2733691c53/CHANGES.rst.

I didn't comb through the code to really validate though (it's a little tough on mobile).

**Comment by sijis** [comment]
I'm closing this in favor of https://github.com/errbotio/errbot/pull/1387, which addresses a lot of changes to the sentry sdk along with flask changes in newer versions.

**Pull request closed without merging**